tests/foreman/virtwho/api/test_nutanix.py

```diff
@@ -197,3 +197,160 @@ def test_positive_hypervisor_id_option(
+            command, form_data['hypervisor_type'], debug=True, org=default_org.label
+        )
+        # Check the option "prism_central=true" should be set in etc/virt-who.d/virt-who.conf
+        config_file = get_configure_file(virtwho_config.id)
+        assert get_configure_option("prism_central", config_file) == 'true'
+        virt_who_instance = (
+            target_sat.api.VirtWhoConfig()
+            .search(query={'search': f'name={virtwho_config.name}'})[0]
+            .status
+        )
+        assert virt_who_instance == 'ok'
+        hosts = [
+            (
+                hypervisor_name,
+                f'product_id={settings.virtwho.sku.vdc_physical} and type=NORMAL',
+            ),
+            (
+                guest_name,
+                f'product_id={settings.virtwho.sku.vdc_physical} and type=STACK_DERIVED',
+            ),
+        ]
+        for hostname, sku in hosts:
+            host = target_sat.cli.Host.list({'search': hostname})[0]
+            subscriptions = target_sat.cli.Subscription.list(
+                {'organization': default_org.name, 'search': sku}
+            )
+            vdc_id = subscriptions[0]['id']
+            if 'type=STACK_DERIVED' in sku:
+                for item in subscriptions:
+                    if hypervisor_name.lower() in item['type']:
+                        vdc_id = item['id']
+                        break
+            target_sat.api.HostSubscription(host=host['id']).add_subscriptions(
+                data={'subscriptions': [{'id': vdc_id, 'quantity': 'Automatic'}]}
+            )
+            result = target_sat.api.Host().search(query={'search': hostname})[0].read_json()
+            assert result['subscription_status_label'] == 'Fully entitled'
+        virtwho_config.delete()
+        assert not target_sat.api.VirtWhoConfig().search(
+            query={'search': f"name={form_data['name']}"}
+        )
+
+    @pytest.mark.tier2
+    def test_positive_prism_central_deploy_configure_by_script(
+        self, default_org, form_data, target_sat
+    ):
+        """Verify "GET /foreman_virt_who_configure/api/ on nutanix prism central mode
+
+        v2/configs/:id/deploy_script"
+
+        :id: d139d70b-7c4a-4325-a892-4829c83755cd
+
+        :expectedresults:
+            Config can be created and deployed on nutanix prism central mode
+            The prism_central has been set in /etc/virt-who.d/vir-who.conf file
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        form_data['prism_flavor'] = "central"
+        virtwho_config = target_sat.api.VirtWhoConfig(**form_data).create()
+        assert virtwho_config.status == 'unknown'
+        script = virtwho_config.deploy_script()
+        hypervisor_name, guest_name = deploy_configure_by_script(
+            script['virt_who_config_script'],
+            form_data['hypervisor_type'],
+            debug=True,
+            org=default_org.label,
+        )
```

Comment:
same can be done here https://github.com/SatelliteQE/robottelo/pull/10016#discussion_r967307813

Reply:
> same can be done here [#10016 (comment)](https://github.com/SatelliteQE/robottelo/pull/10016#discussion_r967307813)
@shubhamsg199 I really totally agree with you, I found there are some other redundant code in test cases, not only exist in nutanix test cases, also exist in other hypervisor, next I will submit another optimization PR, really thanks your good suggestion. I have already develop the update in local, when complete the cases automation will update the changes totally.
As the comments in #10022 reply on https://github.com/SatelliteQE/robottelo/pull/10022#discussion_r965437106

Reply:
> same can be done here [#10016 (comment)](https://github.com/SatelliteQE/robottelo/pull/10016#discussion_r967307813)

@shubhamsg199 Thanks for your good suggestion, have already parametrizing the test cases using @pytest.mark.parametrize('deploy_type', ['id', 'script']), could you please help to review and merge the PR? Many thanks

> same can be done here [#10016 (comment)](https://github.com/SatelliteQE/robottelo/pull/10016#discussion_r967307813)

@shubhamsg199 Thanks for your good suggestion, have already parametrizing the test cases using @pytest.mark.parametrize('deploy_type', ['id', 'script']), could you please help to review and merge the PR? Many thanks